flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotSelectionStrategy.java

```diff
@@ -55,15 +57,15 @@ final class SlotInfoAndResources {
 
         @Nonnull private final ResourceProfile remainingResources;
 
-        private final double taskExecutorUtilization;
+        @Nonnull private final Function<ResourceID, Double> taskExecutorUtilizationLookup;
 
         public SlotInfoAndResources(
```

Comment:
I think we can directly remove the `SlotInfoAndResources` and replace it with `SlotInfoWithUtilization`. This two class has the same functions.  (this change is better to be a separate commit.)